src/app/core/export/download-service/download.service.ts

```diff
@@ -90,17 +90,59 @@ export class DownloadService {
    * @returns string a valid CSV string of the input data
    */
   async createCsv(data: any[]): Promise<string> {
-    // Collect all properties because papa only uses the properties of the first object
+    let entityConstructor: any;
+
+    if (data.length > 0 && typeof data[0]?.getConstructor === 'function') {
+        entityConstructor = data[0].getConstructor();
+    }
     const keys = new Set<string>();
     data.forEach((row) => Object.keys(row).forEach((key) => keys.add(key)));
 
     data = data.map(transformToReadableFormat);
+   
+    if (!entityConstructor) {
+      return this.papa.unparse(data, {
+        quotes: true,
+        header: true,
+        newline: DownloadService.SEPARATOR_ROW,
+        columns: [...keys],
+      });
+    }
+
+    const result = this.exportFile(data, entityConstructor);
+    return result;
+  }
+
+  exportFile(data: any[], entityConstructor: { schema: any; }) {
+    const entitySchema = [entityConstructor.schema];
+    const columnLabel = {};
+
+    entitySchema[0].forEach((value: { value: string, label: string }, key: string) => {
```

Comment:
```suggestion
    const entitySchema = entityConstructor.schema;
    const columnLabel = {};

    entitySchema.forEach((value: { value: string, label: string }, key: string) => {
```

A bit simpler - or was there a special reason for this?